src/rust/src/merkle_frontier.rs

```diff
@@ -98,17 +99,45 @@ pub(crate) struct OrchardWallet;
 
 impl Orchard {
     /// Appends the note commitments in the given bundle to this frontier.
-    pub(crate) fn append_bundle(&mut self, bundle: &orchard_bundle::Bundle) -> bool {
+    pub(crate) fn append_bundle(
+        &mut self,
+        bundle: &orchard_bundle::Bundle,
+    ) -> Result<ffi::OrchardAppendResult, &'static str> {
```

Comment:
Is it possible to return a `Result<Option<[u8; 32]>, &'static str>` here?

Reply:
I don't think so; there's no built-in binding for `Option`.

Reply:
Looks like cxx doesn't support it though. https://github.com/dtolnay/cxx/issues/87